<?xml version="1.0"?>
<sdf version="1.7">
  <model name="blueview_p900">
    <pose>4 0 0.5 0 0.0 3.14</pose>
    <link name="blueview_p900_base_link">
      <inertial>
        <pose>0 0 0 0 0 0</pose>
        <mass>3.5</mass>
        <inertia>
          <ixx>0.0195872</ixx>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyy>0.0195872</iyy>
          <iyz>0</iyz>
          <izz>0.0151357</izz>
        </inertia>
      </inertial>
      <gravity>1</gravity>

      <!-- Camera sensor -->
      <sensor name="camera" type="camera">
        <camera>
          <horizontal_fov>1.047</horizontal_fov>
          <image>
            <width>320</width>
            <height>240</height>
          </image>
          <clip>
            <near>0.1</near>
            <far>100</far>
          </clip>
        </camera>
        <always_on>1</always_on>
        <update_rate>30</update_rate>
        <visualize>true</visualize>
        <topic>/sensor/camera</topic>
      </sensor>

      <!-- Depth Camera sensor -->
      <sensor name="depth_camera" type="depth_camera">
        <update_rate>10</update_rate>
        <topic>/sensor/depth_camera</topic>
        <enable_metrics>true</enable_metrics>
        <camera>
          <horizontal_fov>1.05</horizontal_fov>
          <image>
            <width>320</width>
            <height>240</height>
            <format>R_FLOAT32</format>
          </image>
          <clip>
            <near>0.1</near>
            <far>10.0</far>
          </clip>
        </camera>
      </sensor>

      <!-- Multibeam Sonar sensor -->
      <sensor name="multibeam_sonar" type="custom" gz:type="multibeam_sonar">
        <always_on>true</always_on>
        <update_rate>30.0</update_rate>
        <topic>/sensor/multibeam_sonar</topic>
        <visualize>true</visualize>
        <gz:multibeam_sonar>
        <ray degrees="false">
          <scan>
            <horizontal>
              <beams>512</beams>
              <min_angle>-1.13447</min_angle>
              <max_angle>1.13447</max_angle>
            </horizontal>
            <vertical>
              <rays>300</rays>
              <min_angle>-0.10472</min_angle>
              <max_angle>0.10472</max_angle>
            </vertical>
          </scan>
          <range>
            <min>0.1</min>
            <max>10.0</max>
          </range>
        </ray>
          <spec>
              <!-- Sonar properties -->
            <verticalFOV>12</verticalFOV>
            <sonarFreq>900e3</sonarFreq>
            <bandwidth>29.9e3</bandwidth>
            <soundSpeed>1500</soundSpeed>
            <sourceLevel>220</sourceLevel>
            <maxDistance>10</maxDistance>
            <raySkips>10</raySkips>
            <sensorGain>0.02</sensorGain>
            <blazingSonarImage>true</blazingSonarImage>
            <writeLog>false</writeLog>
            <debugFlag>false</debugFlag>
            <writeFrameInterval>5</writeFrameInterval>
            <!-- ROS publication topics -->
            <pointCloudTopicName>point_cloud</pointCloudTopicName>
            <sonarImageRawTopicName>sonar_image_raw</sonarImageRawTopicName>
            <sonarImageTopicName>sonar_image</sonarImageTopicName>
            <frameName>forward_sonar_optical_link</frameName>
          </spec>
        </gz:multibeam_sonar>
      </sensor>

      <self_collide>0</self_collide>
      <enable_wind>0</enable_wind>
      <kinematic>0</kinematic>

      <visual name="blueview_p900_base_link_visual">
        <pose>0 0 0 0 0 0</pose>
        <geometry>
          <mesh>
            <scale>1 1 1</scale>
            <uri>model://meshes/blueview_p900/p900.dae</uri>
          </mesh>
        </geometry>
        <transparency>0</transparency>
        <cast_shadows>1</cast_shadows>
      </visual>
      <collision name="blueview_p900_base_link_collision">
        <pose>0 0 0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://meshes/blueview_p900/COLLISION-p900.dae</uri>
          </mesh>
        </geometry>
      </collision>
    </link>
    <static>1</static>
    <allow_auto_disable>1</allow_auto_disable>
  </model>
</sdf>
